src/pip/_internal/index.py

```diff
@@ -440,16 +440,59 @@ def evaluate_link(self, link):
         return (True, version)
 
 
+class CandidatePreferences(object):
+
+    """
+    Encapsulates some of the preferences for filtering and sorting
+    InstallationCandidate objects.
+    """
+
+    def __init__(
+        self,
+        prefer_binary=False,  # type: bool
+        allow_all_prereleases=False,  # type: bool
+    ):
+        # type: (...) -> None
+        """
+        :param allow_all_prereleases: Whether to allow all pre-releases.
+        """
+        self.allow_all_prereleases = allow_all_prereleases
+        self.prefer_binary = prefer_binary
+
+
 class CandidateEvaluator(object):
 
     """
     Responsible for filtering and sorting candidates for installation based
     on what tags are valid.
     """
 
+    @classmethod
+    def create(
+        cls,
+        target_python=None,   # type: Optional[TargetPython]
```

Comment:
Out of curiosity, what is the style convention for the offset of these type declaration comments? In some places it is aligned with the others in a declaration, in others there are chunks of aligned comments, and in others they are all just 2 spaces from their corresponding parameters.

Reply:
I don't know of one. I've always been accustomed to two spaces for inline comments, but some people started aligning them. I think it's overboard to always align (e.g. having to change every argument line just because of a change in one line), so I sometimes align to adjacent lines when convenient and it improves the readability / how it looks.